### PR TITLE
Code refactoring

### PR DESCRIPTION
**Pull request by ViachaslauKabak**
Role refactored

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,8 @@ Solr Configuration for SAP Hybris
 =========
 [![License](https://img.shields.io/badge/license-Apache-green.svg?style=flat)](https://raw.githubusercontent.com/lean-delivery/ansible-role-solr-hybris-config/master/LICENSE)
 [![Build Status](https://travis-ci.org/lean-delivery/ansible-role-solr-hybris-config.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-solr-hybris-config)
-[![Galaxy](https://img.shields.io/badge/galaxy-lean__delivery.solr__hybris__config-blue.svg)](https://galaxy.ansible.com/lean_delivery/solr_hybris_config)
+[![Build Status](https://gitlab.com/lean-delivery/ansible-role-solr-hybris-config/badges/master/build.svg)](https://gitlab.com/lean-delivery/ansible-role-solr-hybris-config/pipelines)
+[![Galaxy](https://img.shields.io/badge/galaxy-lean__delivery.solr-hybris-config-blue.svg)](https://galaxy.ansible.com/lean_delivery/solr-hybris-config)
```
Comment (tgadiev) [inline README.md]:
```suggestion
[![Galaxy](https://img.shields.io/badge/galaxy-lean__delivery.solr__hybris__config-blue.svg)](https://galaxy.ansible.com/lean_delivery/solr_hybris_config)
```

File: README.md [hunk 2]
```diff
@@ -12,7 +13,7 @@ This role:
 
 Requirements
 ------------
-  - Minimal Version of the ansible for installation: 2.5
+  - Minimal Version of the ansible for installation: 2.7
   - **Java 8** [![Build Status](https://travis-ci.org/lean-delivery/ansible-role-java.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-java)
   - **Solr installed** [![Build Status](https://travis-ci.org/lean-delivery/ansible-role-solr.svg?branch=master)](https://travis-ci.org/lean-delivery/ansible-role-solr)
   - **Supported OS**:
```
Comment (tgadiev) [inline README.md:19]: Debian?

File: molecule/cloud-aws-direct/molecule.yml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  name: cloud-aws-direct
+  test_sequence:
+    - lint
+    - destroy
+    - dependency
+    - syntax
+    - create
+    - prepare
+    - converge
+#    - idempotence
```
Comment (tgadiev) [inline molecule/cloud-aws-direct/molecule.yml]: Why?

File: molecule/cloud-aws-direct/playbook.yml [hunk 4]
```diff
@@ -3,9 +3,11 @@
   hosts: all
   roles:
     - role: lean_delivery.java
+      transport: openjdk-fallback
```
Comment (tgadiev) [inline molecule/cloud-aws-direct/playbook.yml]: Why do we need this variable in playbook and in inventory at the same time?

File: molecule/cloud-aws-windows/molecule.yml [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+  name: cloud-aws-windows
+  test_sequence:
+    - lint
+    - destroy
+    - dependency
+    - syntax
+    - create
+    - prepare
+    - converge
+#    - idempotence
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]: Why?

File: molecule/cloud-aws-windows/molecule.yml [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        ansible_password: "{{ lookup('env', 'WINRM_ADMIN_PWD') | default('M0leCule123!', True) }}"
+        ansible_winrm_transport: credssp
+        ansible_winrm_server_cert_validation: ignore
+  playbooks:
+    create: ../resources/provisioning/WINRM/create.yml
+    destroy: ../resources/provisioning/WINRM/destroy.yml
+  lint:
+    name: ansible-lint
+  env:
+    ANSIBLE_LIBRARY: ${ANSIBLE_LIBRARY}
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml:46]: Not needed for AWS

File: tasks/main.yml [hunk 7]
```diff
@@ -1,38 +1,16 @@
...
+- name: Load a variables
+  include_vars: '{{ item }}'
   with_first_found:
-    - "fetch/{{ transport }}.yml"
-    - "fetch/unknown-transport.yml"
-  loop_control:
-    loop_var: transport_driver
+    - '{{ ansible_system }}.yml'
+    - '{{ ansible_os_family }}.yml'
+    - 'system/not-supported.yml'
```
Comment (tgadiev) [inline tasks/main.yml]: Is that variable?

File: tasks/system/Linux.yml [hunk 8]
```diff
@@ -1,27 +1,66 @@
 ---
-- name: "Install contrib hybris components"
+- name: 'Install pip package'
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: These pre-requisites should be moved to pre_tasks or to prepare stage. It's not scope of hybris-config role to install boto.

File: tasks/system/Linux.yml [hunk 9]
```diff
@@ -1,27 +1,66 @@
...
+  package:
+    name: python-pip
+    state: present
+  register: status
+  until: status is succeeded
+  become: True
+  when: transport == 's3'
+
+- name: Install boto module
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: These pre-requisites should be moved to pre_tasks or to prepare stage. It's not scope of hybris-config role to install boto.

File: tasks/system/Linux.yml [hunk 10]
```diff
@@ -1,27 +1,66 @@
...
-    creates: "{{ dest_solr_path }}/contrib/hybris"
+    creates: '{{ solr_dest_path }}/contrib/hybris'
+    remote_src: True
   become: True
   notify:
-    - "restart solr linux"
+    - restart Solr Linux
 
-- name: "Install data hybris components"
+- name: Install data hybris components
```
Comment (tgadiev) [inline tasks/system/Linux.yml]: May be combine to one task with previous?

File: tasks/system/Windows.yml [hunk 11]
```diff
@@ -1,30 +1,43 @@
...
   win_copy:
-    src: "{{ solr_contrib_hybris_patch }}"
-    dest: "{{ dest_main_path }}\\"
+    src: '{{ solr_contrib_hybris_patch }}'
+    dest: '{{ solr_dest_main_path }}/'
     force: True
+    remote_src: True
 
-- name: "Upload data hybris components"
+- name: Upload data hybris components
```
Comment (tgadiev) [inline tasks/system/Windows.yml]: May be combine to one task with previous?

File: tasks/system/Windows.yml [hunk 12]
```diff
@@ -1,30 +1,43 @@
...
-    dest: "{{ dest_solr_path }}"
-    creates: "{{ dest_solr_path }}\\contrib\\hybris"
-  notify: "restart solr windows"
+    src: '{{ solr_contrib_hybris_patch }}'
+    dest: '{{ solr_dest_path }}'
+    creates: '{{ solr_dest_path }}/contrib/hybris'
+  notify: restart Solr Windows
 
-- name: "Install data hybris components"
+- name: Install data hybris components
```
Comment (tgadiev) [inline tasks/system/Windows.yml]: May be combine to one task with previous?

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (ViachaslauKabak) [reply README.md:19]: done

File: tasks/system/Windows.yml [hunk 13]
```diff
@@ -1,30 +1,72 @@
...
+    loop_var: hybris_patch_file
+
+- name: Create data temporary directory
+  win_tempfile:
+    state: directory
+    suffix: solr_hybris_data
+  register: hybris_solr_data_temp_dir
+  changed_when: False
+
+- name: Create contrib temporary directory
```
Comment (tgadiev) [inline tasks/system/Windows.yml:29]: Why separate task?

File: molecule/cloud-aws-windows/molecule.yml [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+  options:
+    config-file: .yamllint
+platforms:
+
+  - name: test-aws-Windows2016-solr-hybris-config
+    aws_image_id: ami-00a20f508263efd30
+    aws_region: us-east-1
+    instance_type: m5.large
+    vpc_subnet_id: subnet-05a2ef2b767afec50
+    assign_public_ip: False
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]:
```suggestion
    assign_public_ip: True
```

File: molecule/cloud-aws-windows/molecule.yml [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+  name: yamllint
+  options:
+    config-file: .yamllint
+platforms:
+
+  - name: test-aws-Windows2016-solr-hybris-config
+    aws_image_id: ami-00a20f508263efd30
+    aws_region: us-east-1
+    instance_type: m5.large
+    vpc_subnet_id: subnet-05a2ef2b767afec50
```
Comment (tgadiev) [inline molecule/cloud-aws-windows/molecule.yml]:
```suggestion
    vpc_subnet_id: subnet-01be2149
```

**Review by tgadiev — CHANGES_REQUESTED** [review]

File: tasks/system/Windows.yml (hunk 13 above)
Comment (ViachaslauKabak) [reply tasks/system/Windows.yml:29]: not to harden the role

**Review by tgadiev — APPROVED** [review]

**Pull request merged**